kafka-rest/src/main/java/io/confluent/kafkarest/ConsumerManager.java

```diff
@@ -354,6 +358,58 @@ private ConsumerState getConsumerInstance(String group, String instance) {
     ConsumerConnector createConsumer(ConsumerConfig config);
   }
 
+  class ReadFutureTask<V> extends FutureTask<V> {
+
+    private Runnable myTask;
+
+    private Callable callable;
```

Comment:
I can understand the desire to make this more general, but we only need this to wrap `RunnableReadTask`, which implements `Runnable` and not `Callable`.  I think we can simplify a little bit by removing this field and the `newTaskFor(Callable<T> callable)` override in `KafkaConsumerThreadPoolExecutor` below.

Reply:
thanks